### PR TITLE
chore: add kustomize for easier config adjustments

### PR DESCRIPTION
**Pull request by estahn**
This allows easier management via tools that support kustomize, e.g. ArgoCD.

### CODE REVIEW

File: manifests/setup/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: kubesphere-logging-system
+commonLabels:
```
Comment (benjaminhuo) [inline manifests/setup/kustomization.yaml:5]: Thanks for the PR @estahn 

The common label you add `app.kubernetes.io/name: fluentbit-operator` already exists in the following files:
- `fluentbit-operator-clusterRoleBinding.yaml` 
- `fluentbit-operator-deployment.yaml` 
- `fluentbit-operator-clusterRole.yaml`  
- `fluentbit-operator-serviceAccount.yaml`

And `fluentbit-operator-deployment.yaml` and `fluentbit-operator-serviceAccount.yaml` already have namespace `kubesphere-logging-system`

What you add here is a placeholder and maybe changed to other label or namespaces?

Maybe it's not necessary to add label `app.kubernetes.io/name: fluentbit-operator` to `namespace-namespace.yaml` and `fluentbit-operator-crd.yaml`

**Review by benjaminhuo — COMMENTED** [review]
Refer to my comments above

Thanks
Ben

File: manifests/setup/kustomization.yaml (hunk 1 above)
Comment (estahn) [reply manifests/setup/kustomization.yaml]: Good points. The adjustments should be made downstream. I also removed the namespace resource as this is something that should done downstream too.

File: manifests/setup/kustomization.yaml [hunk 2]
```diff
@@ -1,12 +1,7 @@
...
 kind: Kustomization
 
-namespace: kubesphere-logging-system
-commonLabels:
-  app.kubernetes.io/name: fluentbit-operator
-
 resources:
-- namespace-namespace.yaml
 - fluentbit-operator-clusterRoleBinding.yaml
 - fluentbit-operator-deployment.yaml
```
Comment (benjaminhuo) [inline manifests/setup/kustomization.yaml:6]: There is nothing kustomized in the new commit now, do we still need this？

Comment (estahn) [reply manifests/setup/kustomization.yaml:6]: Yes, this is still needed. All adjustments happen downstream, but require a kustomize file upstream, e.g.

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - github.com/sighupio/gatekeeper-policy-manager?ref=v0.4.1
  - ingress.yaml

patchesStrategicMerge:
  - set-memory.yaml
```

File: manifests/setup/kustomization.yaml [hunk 3]
```diff
@@ -1,12 +1,7 @@
...
 - fluentbit-operator-crd.yaml
```
Comment (benjaminhuo) [inline manifests/setup/kustomization.yaml:7]: The crd needn't be customized, I would suggest the resources includes below 2 files:

fluentbit-operator-deployment.yaml
fluentbit-operator-serviceAccount.yaml

**Review by benjaminhuo — CHANGES_REQUESTED** [review]
See my comments above

**Comment by estahn** [comment]
The CRDs will need to be deployed as well, regardless of whether they get
customised.

**Comment by benjaminhuo** [comment]
Got you, actually what you need is a placeholder. And you'll customize it as you wish.

**Pull request merged**